core/encoding/hex/hex.odin

```diff
@@ -0,0 +1,74 @@
+package hex
+
+import "core:strings"
+
+encode :: proc(src: []byte, allocator := context.allocator) -> []byte #no_bounds_check {
+	dst := make([]byte, len(src) * 2, allocator)
+    for i := 0; i < len(src); i += 1 {
+		v := src[i]
+        dst[i]   = HEXTABLE[v>>4]
+        dst[i+1] = HEXTABLE[v&0x0f]
+        i += 2
+    }
+
+	return dst
+}
+
+
+decode :: proc(src: []byte, allocator := context.allocator) -> (dst: []byte, ok: bool) #no_bounds_check {
+	if len(src) % 2 == 1 {
+		return
+	}
+
+	dst = make([]byte, len(src) / 2, allocator)
+	for i, j := 0, 1; j < len(src); j += 2 {
+		p := src[j-1]
+		q := src[j]
+
+		a := hex_digit(p) or_return
+		b := hex_digit(q) or_return
+
+		dst[i] = (a << 4) | b
+		i += 1
+	}
+
+	return dst, true
+}
+
+// Decodes the given sequence into one byte.
+// Should be called with one rune worth of the source, eg: 0x23 -> '#'.
+decode_sequence :: proc(str: string) -> (byte, bool) {
```

Comment:
This can be simplified a bit if we take advantage of the sequence having a length of 2 without the prefix. And while we're at it, let's also handle the uppercase prefix seeing as we handle uppercase hex digits.

```odin
decode_sequence :: proc(str: string) -> (res: byte, ok: bool) {
	str := str
	if strings.has_prefix(str, "0x") || strings.has_prefix(str, "0X") {
		str = str[2:]
	}

	if len(str) != 2 {
		return 0, false
	}

	upper := hex_digit(str[0]) or_return
	lower := hex_digit(str[1]) or_return

	return upper << 4 | lower, true
}
```